### PR TITLE
Nightly: Fix issues on test.

### PR DESCRIPTION
**Pull request by eloycoto**
Fix issues on builds:

https://jenkins.cilium.io/job/Cilium-Master-Nightly-Tests-All/207/
https://jenkins.cilium.io/job/Cilium-Master-Nightly-Tests-All/206/
https://jenkins.cilium.io/job/Cilium-Master-Nightly-Tests-All/205/

Test didn't wait until pods are ready.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4123)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-nightly

**Comment by ianvernon** [comment]
@eloycoto please amend the commit to be more descriptive of what the actual fixes are. The links will be dead after Jenkins GC's them, and thus will be useless as time goes on. Furthermore, the commit should explain what the fix is explicitly.

File: test/helpers/policygen/policygen.go
Comment (ianvernon) [inline test/helpers/policygen/policygen.go]: What is the need for these case changes?

File: test/helpers/policygen/models.go
Comment (ianvernon) [inline test/helpers/policygen/models.go:796]: What is the exact reasoning for this change? It is not clear.

**Review by ianvernon — CHANGES_REQUESTED** [review]
Please fix the commit message as outlined above. Also please answer my questions inline.

File: test/helpers/policygen/models.go
Comment (eloycoto) [reply test/helpers/policygen/models.go:796]: The main reason is to create pods and the given service at the same time. It didn't make sense to me to have a function called `ConnectivityTest`  that created some services. This is the way that we follow in Nightly.

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-nightly

**Review by ianvernon — APPROVED** [review]

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-me-please

**Review by raybejjani — APPROVED** [review]

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
Found the fails a bit suspicious, that maybe are related with #4234. This is why
I think that it is related:

Cilium tunnel list on k8s4
```
TUNNEL            VALUE
f00d::a0f:0:0:0   192.168.36.14
10.10.0.0         192.168.36.11
10.15.0.0         192.168.36.14
f00d::a0a:0:0:0   192.168.36.11
```

Cilium health status in k8s4

```
Probe time:   2018-05-25T14:22:02Z
Nodes:
  k8s4 (localhost):
    Host connectivity to 192.168.36.14:
      ICMP:   OK, RTT=218.92µs
```


On k8s1

Tunnel list:
```
TUNNEL            VALUE
f00d::a0f:0:0:0   192.168.36.14
10.10.0.0         192.168.36.11
10.15.0.0         192.168.36.14
f00d::a0a:0:0:0   192.168.36.11

```
Health:

```
Probe time:   2018-05-25T14:22:06Z
Nodes:
  k8s1 (localhost):
    Host connectivity to 192.168.36.11:
      ICMP:          OK, RTT=841.611µs
      HTTP via L3:   OK, RTT=2.086421ms
    Endpoint connectivity to 10.10.224.195:
      ICMP:          OK, RTT=930.961µs
      HTTP via L3:   OK, RTT=1.4904ms
  k8s2:
    Host connectivity to 192.168.36.12:
      ICMP:          OK, RTT=7.225322ms
      HTTP via L3:   OK, RTT=2.712137ms
    Endpoint connectivity to 10.15.55.20:
      ICMP:          Connection timed out
      HTTP via L3:   Connection timed out
  k8s3:
    Host connectivity to 192.168.36.13:
      ICMP:          OK, RTT=954.498µs
      HTTP via L3:   OK, RTT=3.484592ms
    Endpoint connectivity to 10.15.227.162:
      ICMP:          Connection timed out
      HTTP via L3:   Connection timed out
  k8s4:
    Host connectivity to 192.168.36.14:
      ICMP:          OK, RTT=1.727189ms
      HTTP via L3:   OK, RTT=3.685799ms
    Endpoint connectivity to 10.15.99.122:
      ICMP:          OK, RTT=7.225376ms
      HTTP via L3:   OK, RTT=3.437147ms

```

K8s2:

```
TUNNEL            VALUE
f00d::a0f:0:0:0   192.168.36.12
10.10.0.0         192.168.36.11
10.15.0.0         192.168.36.12
f00d::a0a:0:0:0   192.168.36.11
```


```
Probe time:   2018-05-25T14:22:05Z
Nodes:
  k8s2 (localhost):
    Host connectivity to 192.168.36.12:
      ICMP:   OK, RTT=1.732364ms
    Endpoint connectivity to 10.15.55.20:
      ICMP:   OK, RTT=1.895757ms
  k8s1:
    Host connectivity to 192.168.36.11:
      ICMP:   OK, RTT=1.91744ms
    Endpoint connectivity to 10.10.224.195:
      ICMP:   Connection timed out
  k8s3:
    Host connectivity to 192.168.36.13:
      ICMP:   OK, RTT=1.901145ms
    Endpoint connectivity to 10.15.227.162:
      ICMP:   Connection timed out
  k8s4:
    Host connectivity to 192.168.36.14:
      ICMP:   OK, RTT=1.968047ms
    Endpoint connectivity to 10.15.99.122:
      ICMP:   Connection timed out
```


Logs in cilium-drjxc
```
2018-05-25T14:20:54.140951089Z level=info msg="Automatically retrieved IP for node. Using it for ipv4-node" ipAddr=192.168.36.14 node=k8s4
2018-05-25T14:20:54.140954687Z level=info msg="Initializing node addressing"
2018-05-25T14:20:54.140958383Z level=info msg="Using autogenerated IPv4 allocation range" v4Prefix=10.15.0.0/16
2018-05-25T14:20:54.140983698Z level=info msg="Using autogenerated IPv6 allocation range" v6Prefix="f00d::a0f:0:0:0/96"
```

Logs in cilium-l4gh5


```
2018-05-25T14:20:53.363514752Z level=info msg="Received own node information from API server" ipAddr.ipv4=192.168.36.11 ipAddr.ipv6="<nil>" nodeName=k8s1 subsys=k8s
2018-05-25T14:20:53.363528429Z level=info msg="Retrieved IPv4 allocation range for node. Using it for ipv4-range" node=k8s1 v4Prefix=10.10.0.0/16
2018-05-25T14:20:53.363532871Z level=info msg="Automatically retrieved IP for node. Using it for ipv4-node" ipAddr=192.168.36.11 node=k8s1
2018-05-25T14:20:53.36353665Z level=info msg="Initializing node addressing"
```

Next steps:

- Wait until #4219 get merged (Already changed the Cilium-ds.yaml)
- Rebase this one and trigger the nightly again.

Logs: 
[10cd9672_NightlyPolicies_PolicyEnforcement_default_L3_No_Policy_L4_No_Policy_L7_No_Policy_Destination_service(1).zip](https://github.com/cilium/cilium/files/2039888/10cd9672_NightlyPolicies_PolicyEnforcement_default_L3_No_Policy_L4_No_Policy_L7_No_Policy_Destination_service.1.zip)

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-nightly

**Comment by eloycoto** [comment]
test-me-please

**Pull request merged**
